examples/src/bin/server_acceptor.rs

```diff
@@ -0,0 +1,336 @@
+        // thread.
+        let mut crl_file = File::open(crl_path).unwrap();
+        let mut crl = Vec::default();
+        crl_file.read_to_end(&mut crl).unwrap();
+
+        // Construct a fresh verifier using the test PKI roots, and the updated CRL.
+        let verifier = WebPkiClientVerifier::builder(self.roots.clone())
+            .with_crls([UnparsedCertRevocationList(crl)])
+            .build()
+            .unwrap();
+
+        // Build a server config using the fresh verifier. If necessary, this could be customized
+        // based on the ClientHello (e.g. selecting a different certificate, or customizing
+        // supported algorithms/protocol versions).
+        Arc::new(
+            ServerConfig::builder()
+                .with_safe_defaults()
+                .with_client_cert_verifier(verifier)
+                .with_single_cert(
+                    vec![Certificate(self.server_cert_der.clone())],
+                    PrivateKey(self.server_key_der.clone()),
+                )
+                .unwrap(),
+        )
+    }
+
+    /// Issue a certificate revocation list (CRL) for the revoked `serials` provided (may be empty).
+    /// The CRL will be signed by the test PKI CA and returned in DER serialized form.
+    fn crl(&self, serials: Vec<rcgen::SerialNumber>, next_update_seconds: u64) -> Vec<u8> {
+        // In a real use-case you would want to set this to the current date/time.
+        let now = rcgen::date_time_ymd(2023, 1, 1);
+
+        // For each serial, create a revoked certificate entry.
+        let revoked_certs = serials
+            .into_iter()
+            .map(|serial| rcgen::RevokedCertParams {
+                serial_number: serial,
+                revocation_time: now,
+                reason_code: Some(rcgen::RevocationReason::KeyCompromise),
+                invalidity_date: None,
+            })
+            .collect();
+
+        // Create a new CRL signed by the CA cert.
+        let crl = rcgen::CertificateRevocationListParams {
+            this_update: now,
+            next_update: now.add(Duration::from_secs(next_update_seconds)),
+            crl_number: rcgen::SerialNumber::from(1234),
+            revoked_certs,
+            key_identifier_method: rcgen::KeyIdMethod::Sha256,
+            alg: &rcgen::PKCS_ECDSA_P256_SHA256,
+        };
+        rcgen::CertificateRevocationList::from_params(crl)
+            .unwrap()
+            .serialize_der_with_signer(&self.ca_cert)
+            .unwrap()
+    }
+}
+
+/// CRL updater function that runs in a separate thread. This periodically updates the CRL file
+/// on disk, flipping between writing a CRL that describes the client certificate as revoked, and
+/// a CRL that describes the client certificate as not revoked.
+///
+/// In a real use case, the CRL would be updated by fetching fresh CRL data from an authoritative
+/// distribution point.
+fn crl_updater(duration: Duration, crl_path: &Path, pki: Arc<TestPki>) {
```

Comment:
Style suggestion: for these kinds of little thread thingies, I like to write a `struct` definition with the input state it needs, with a `run(self)` method on it. I feel like that more clearly signals intent that this is not just a normal function.